### PR TITLE
Switch all data files to Zstandard compression

### PR DESCRIPTION
**Pull request by clavin-xlnx**

### CODE REVIEW

File: src/com/xilinx/rapidwright/util/FileTools.java [hunk 1]
```diff
@@ -1672,6 +1698,23 @@ public static boolean isDataBinary(InputStream is) {
...
+        
+    public static boolean isFileGzipped(Path path) {
+        try (InputStream in = Files.newInputStream(path)) {
+            boolean isGzipped = false;
+            try {
+                isGzipped = in.read() == GZIP_START_BYTES[0] 
+                         && in.read() == GZIP_START_BYTES[1];
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/util/FileTools.java:1711]: Do you need this inner `try/catch`?

Comment (clavin-xlnx) [reply src/com/xilinx/rapidwright/util/FileTools.java]: Thanks, I've simplified it.

**Review by eddieh-xlnx — APPROVED** [review]

**Comment by clavin-xlnx** [comment]
To summarize the impact of this change for the data files, device model file sizes were reduced by a geomean of 32% with a geomean reduction in load time by 11%.  For the device node cache files, the savings were even more pronounced, 52% smaller file sizes with a 108% reduction in load times.

image

image

image

image

**Comment by mithro** [comment]
QQ - Did you do any test comparing SSD verse spinning rust? I guess it would be even more pronounced / better on slower media.

**Comment by clavin-xlnx** [comment]
> QQ - Did you do any test comparing SSD verse spinning rust? I guess it would be even more pronounced / better on slower media.

We didn't try testing specific storage technologies.  These tests were all on network attached storage (not sure of the physical hardware) which is fairly robust when not under significant load.  We did some similar tests with a RAMDisk and found the performance mostly the same.  But yes, presumably the improvement should be more significant on slower storage.

**Pull request merged**